airflow/www/static/js/api/useDagCode.ts

```diff
@@ -0,0 +1,98 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import axios, { AxiosResponse } from "axios";
+import { useQuery } from "react-query";
+import { useRef, useEffect, useState } from "react";
+
+import { getMetaValue } from "src/utils";
+import type { API } from "src/types";
+import useErrorToast from "../utils/useErrorToast";
+
+interface DagCodeData {
+  lastParsedTime?: string;
+  codeSource?: string;
+}
+
+export default function useDagCode() {
+  const errorToast = useErrorToast();
+  const [data, setData] = useState<DagCodeData>({
+    lastParsedTime: "N/A",
+    codeSource: "",
+  });
+
+  const dagQuery = useQuery(
+    ["dagQuery"],
+    () => {
+      const dagApiUrl = getMetaValue("dag_api");
+      return axios.get<AxiosResponse, API.DAG>(dagApiUrl);
+    },
+    {
+      onError: (error: Error) => {
+        errorToast({
+          error,
+        });
+      },
+    }
+  );
+
+  const dagSourceQuery = useQuery(
+    ["dagSourceQuery"],
+    () => {
+      const fileToken = dagQuery.data?.fileToken || "";
+      const dagSourceApiUrl = getMetaValue("dag_source_api").replace(
+        "_FILE_TOKEN_",
+        fileToken
+      );
+      return axios.get<AxiosResponse, string>(dagSourceApiUrl, {
+        headers: { Accept: "text/plain" },
+      });
+    },
+    {
+      enabled: false,
```

Comment:
```suggestion
      enabled: !!dagQuery.data.fileToken
```

And then we can return the entire dagSourceQuery and remove all of the `useEffect`s

Reply:
Documentation: https://tanstack.com/query/v3/docs/react/guides/dependent-queries

Reply:
I can use the code suggestion to remove the first `useEffects` function. 
I will still need the second one as I am pulling the `last_parsed_date` column from `dagApi` and the `dag_code` from `dagSourceApi` and need to combine them before returning. Please let me know if this needs to change.